devlib/utils/gem5.py

```diff
@@ -39,5 +46,6 @@ def iter_statistics_dump(stats_file):
             if res:
                 k = res.group("key")
                 v = res.group("value").split()
+                v = map(best_effort_numeric, v)
```

Comment:
I missed your 3rd paragraph ...
What about modifying the `catch` part of `best_effort_numeric` to log the issue before returning the value ?